### ISSUE TITLE
Stack overflow

### ISSUE THREAD

**Issue by kevinresol** [HaxeFoundation/haxe#6216]
Haxe 3.4.2

```haxe
import tink.testrunner.*;

class Main {
	static function main() {
		var a:Assertions = [new Assertion(true, '')];
	}
}
```

Then,

```
haxelib install tink_testrunner
haxe -cp src -main Main -lib tink_testrunner
```

Gives a simple `Stack overflow` output.

With `OCAMLRUNPARAM=b`: 

```
Fatal error: exception Stack_overflow
Called from file "list.ml", line 84, characters 24-34
Called from file "src/typing/typer.ml", line 3302, characters 95-120
Called from file "src/typing/typer.ml", line 3302, characters 46-52
Called from file "list.ml", line 84, characters 24-34
Called from file "src/typing/typer.ml", line 3302, characters 11-120
Called from file "src/typing/typer.ml", line 3302, characters 46-52
Called from file "list.ml", line 84, characters 24-34
Called from file "src/typing/typer.ml", line 3302, characters 11-120
Called from file "src/typing/typer.ml", line 3302, characters 46-52
Called from file "list.ml", line 84, characters 24-34
...
```

**Comment by Simn** [comment]
This probably has to do with `get_abstract_froms`, but the stack trace is completely useless... Not sure what OCaml does there exactly.

**Comment by benmerckx** [comment]
Something similar happens when trying to compile [tink_sql](https://github.com/haxetink/tink_sql) (pure branch) to php or php7:
````
Called from file "list.ml", line 84, characters 24-34
Called from file "src/typing/typer.ml", line 2845, characters 11-124
Called from file "src/typing/typer.ml", line 2845, characters 46-57
Called from file "list.ml", line 84, characters 24-34
Called from file "src/typing/typer.ml", line 2845, characters 11-124
Called from file "src/typing/typer.ml", line 2845, characters 46-57
````
`-v` stops at `StdDriver.StdConnection.selectAll`:
````
Typing tink.sql.drivers.sys.StdConnection.selectAll
Stack overflow
````

**Comment by RealyUniqueName** [comment]
Fixed by #8592
